### PR TITLE
Refactor(eos_designs): Return list-based data models from eos_designs python modules - MLAG

### PR DESCRIPTION
**Pull request by ghost**
## Related Issue(s)

Fixes #<ISSUE ID>

## Component(s) name

`arista.avd.eos_designs`

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/utils_filtered_tenants.py [hunk 1]
```diff
@@ -260,7 +260,7 @@ def _filtered_svis(self, vrf: dict) -> list[dict]:
         Filtering based on accepted vlans since eos_designs_facts already
         filtered that on tags and trunk_groups.
         """
-        svis: list[dict] = natural_sort(convert_dicts(vrf.get("svis", []), "id"), "id")
+        svis = vrf.get("svis", [])
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/utils_filtered_tenants.py]: No need to change this line. It already took a list, but it also contained the conversions if needed. Please revert this change.

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/vlans.py [hunk 2]
```diff
@@ -27,23 +27,34 @@ def vlans(self) -> dict | None:
         The function also creates uplink_native_vlan for this switch or downstream switches.
         """
 
-        vlans = {}
+        vlans = []
         # TODO - can probably do this with sets but need list in the end so not sure it is worth it
         for vlan_trunk_group in self._underlay_vlan_trunk_groups:
             for vlan in range_expand(vlan_trunk_group["vlan_list"]):
-                vlans.setdefault(int(vlan), {"trunk_groups": []})["trunk_groups"].extend(vlan_trunk_group["trunk_groups"])
+                if (target := get_item(vlans, "id", int(vlan))) is None:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/vlans.py]: I don't think the name `target` works well here. Maybe `found_vlan`.

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/vlans.py [hunk 3]
```diff
@@ -27,23 +27,34 @@ def vlans(self) -> dict | None:
...
+                    vlan_dict = {
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/vlans.py]: Maybe to keep it consistent, use `new_vlan` instead of `vlan_dict` here.

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/vlans.py [hunk 4]
```diff
@@ -27,23 +27,34 @@ def vlans(self) -> dict | None:
...
+                        "id": int(vlan),
+                        "trunk_groups": [],
+                    }
+                    vlan_dict["trunk_groups"].extend(vlan_trunk_group["trunk_groups"])
+                    vlans.append(vlan_dict)
+                else:
+                    for trunk_group in vlan_trunk_group["trunk_groups"]:
+                        if trunk_group not in target["trunk_groups"]:
+                            target["trunk_groups"].extend(vlan_trunk_group["trunk_groups"])
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/vlans.py]: Something looks wrong here. You are looping over each trunk_group but extending the full list for every loop.
I suggest to just extend it like the old code did, since we remove duplicated below by converting to `set`

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests_v4.0/intended/structured_configs/DC1-CL1A.yml [hunk 5]
```diff
@@ -241,6 +241,16 @@ management_api_http:
   enable_https: true
   default_services: false
 vlans:
+- id: 4090
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests_v4.0/intended/structured_configs/DC1-CL1A.yml:244]: These changes are OK, since eos_designs_unit_tests_v4.0 does not have any `convert_dict` functionality, so it breaks when we merge inside `yaml_templates_to_facts`.

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/utils_filtered_tenants.py (hunk 1 above)
Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/utils_filtered_tenants.py]: Reverted

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/vlans.py (hunk 2 above)
Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/vlans.py]: fixed

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/vlans.py (hunk 3 above)
Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/vlans.py]: fixed

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/vlans.py (hunk 4 above)
Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/vlans.py]: fixed

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/mlag/avdstructuredconfig.py [hunk 6]
```diff
@@ -34,20 +34,26 @@ def spanning_tree(self):
         return {"no_spanning_tree_vlan": self.shared_utils.mlag_peer_vlan}
 
     @cached_property
-    def vlans(self):
-        vlans = {}
+    def vlans(self) -> list | None:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/mlag/avdstructuredconfig.py]:
```suggestion
    def vlans(self) -> list:
```

**Review by ClausHolbechArista — APPROVED** [review]

**Review by carlbuchmann — APPROVED** [review]
LGTM!
